api/queries_pr.go

```diff
@@ -300,6 +349,27 @@ func (pr *PullRequest) ChecksStatus() PullRequestChecksStatus {
 	return summary
 }
 
+type checkStatus int
+
+const (
+	passing checkStatus = iota
+	failing
+	pending
+)
+
+func parseCheckStatus(state string) checkStatus {
```

Comment:
Probably makes sense to reuse this logic for the old state/conclusion work as well.